BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsClient.kt

```diff
@@ -0,0 +1,220 @@
+            WORK_INPUT_KEY_EVENT_NAME to eventName,
+            WORK_INPUT_KEY_TIMESTAMP to timestamp
+        )
+        val analyticsWorkRequest =
+            OneTimeWorkRequest.Builder(AnalyticsWriteToDbWorker::class.java)
+                .setInputData(inputData)
+                .build()
+        workManager.enqueueUniqueWork(
+            WORK_NAME_ANALYTICS_WRITE, ExistingWorkPolicy.APPEND_OR_REPLACE, analyticsWorkRequest
+        )
+    }
+
+    fun writeAnalytics(inputData: Data): ListenableWorker.Result {
+        val eventName = inputData.getString(WORK_INPUT_KEY_EVENT_NAME)
+        val timestamp = inputData.getLong(WORK_INPUT_KEY_TIMESTAMP, INVALID_TIMESTAMP)
+
+        return if (eventName == null || timestamp == INVALID_TIMESTAMP) {
+            ListenableWorker.Result.failure()
+        } else {
+            val event = AnalyticsEvent(eventName, timestamp)
+            val analyticsEventDao = analyticsDatabase.analyticsEventDao()
+            analyticsEventDao.insertEvent(event)
+            ListenableWorker.Result.success()
+        }
+    }
+
+    private fun scheduleAnalyticsUpload(
+        configuration: Configuration,
+        authorization: Authorization,
+        sessionId: String?,
+        integration: String?
+    ): UUID {
+        val inputData = workDataOf(
+            WORK_INPUT_KEY_AUTHORIZATION to authorization.toString(),
+            WORK_INPUT_KEY_CONFIGURATION to configuration.toJson(),
+            WORK_INPUT_KEY_SESSION_ID to sessionId,
+            WORK_INPUT_KEY_INTEGRATION to integration
+        )
+        val analyticsWorkRequest = OneTimeWorkRequest.Builder(AnalyticsUploadWorker::class.java)
+            .setInitialDelay(30, TimeUnit.SECONDS)
+            .setInputData(inputData)
+            .build()
+        workManager.enqueueUniqueWork(
+            WORK_NAME_ANALYTICS_UPLOAD, ExistingWorkPolicy.KEEP, analyticsWorkRequest
+        )
+        return analyticsWorkRequest.id
+    }
+
+    fun uploadAnalytics(context: Context?, inputData: Data): ListenableWorker.Result {
+        val configuration = getConfigurationFromData(inputData)
+        val authorization = getAuthorizationFromData(inputData)
+        val sessionId = inputData.getString(WORK_INPUT_KEY_SESSION_ID)
+        val integration = inputData.getString(WORK_INPUT_KEY_INTEGRATION)
+        val isMissingInputData =
+            listOf(configuration, authorization, sessionId, integration).contains(null)
+        return if (isMissingInputData) {
+            ListenableWorker.Result.failure()
+        } else try {
+            val analyticsEventDao = analyticsDatabase.analyticsEventDao()
+            val events = analyticsEventDao.getAllEvents()
+            if (events.isNotEmpty()) {
+                val metadata = deviceInspector.getDeviceMetadata(context, sessionId, integration)
+                val analyticsRequest = serializeEvents(authorization, events, metadata)
+                val analyticsUrl = configuration!!.analyticsUrl
```

Comment:
`!!` could throw a NPE right?

Reply:
Yup! Good call definitely a side effect of using the automated Kotlin conversion tool. I'll update.